File: tc-001/checkAge.js
const readline = require("readline");

// Basic variables
const majorityAge = 18;
const minMajorityYear = 2024 - majorityAge; // 2006

function checkAge(birthYear) {
  if (birthYear <= minMajorityYear) {
    console.log("You are an adult!");
  } else {
    const yearsToMajority = birthYear - minMajorityYear ;
    console.log(`You are not an adult! You have ${yearsToMajority} years left until majority!`
    );
  }
}

// Create an interface for reading input from the console
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

// Prompt the user for birth year
rl.question("Enter your birth year: ", (answer) => {
  const birthYear = parseInt(answer, 10);
  checkAge(birthYear);
  rl.close();
});

File: tc-001/checkAge.c
#include <stdio.h>

int main() {
    int majority_age = 18;               // Simple assigned value
    int min_majority_year = 2024 - 18;   // Dynamically assigned value
    int birth_year;

    printf("Enter your birth year: ");
    scanf("%d", &birth_year);

    // Conditional
    if (birth_year <= min_majority_year) {
        printf("You are an adult!\n");
    } else {
        int years_to_majority = birth_year - min_majority_year ;
        printf("You are not an adult! You have %d years left until majority!\n", years_to_majority);
    }

    return 0;
}
